### ISSUE TITLE
Reconfigure mobile testing to use 4G speeds

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/httparchive#138]
Data from the Chrome UX Report indicates that the vast majority of connection speeds on mobile devices (phone/tablet) are effectively 4G:

```sql
#standardSQL
SELECT
  SUM(IF(form_factor.name IN ('phone', 'tablet') AND effective_connection_type.name = '4G', bin.density, 0))
    / SUM(IF(form_factor.name IN ('phone', 'tablet'), bin.density, 0)) AS pct_mobile_4g
FROM 
  `chrome-ux-report.chrome_ux_report.201710`,
  UNNEST(first_contentful_paint.histogram.bin) AS bin
```

https://bigquery.cloud.google.com/results/chrome-ux-report:bquijob_35020ffa_160510ab375

**Result:** 87.95%

Update the WebPageTest configuration for mobile tests to use 4G speeds to more accurately represent real user conditions.

**Comment by igrigorik** [comment]
+1. Ideally, this should be coupled with similar default updates in WebPageTest and Lighthouse emulation. /cc @pmeenan @paulirish

**Comment by pmeenan** [comment]
QQ.  Since CrUX equalizes all origins, does that query basically average across all origins?

Not sure it changes much, but looking at traffic to Google is 90.10% (assuming my query is correct):

```sql
#standardSQL
SELECT
  SUM(IF(form_factor.name IN ('phone', 'tablet') AND effective_connection_type.name = '4G', bin.density, 0))
    / SUM(IF(form_factor.name IN ('phone', 'tablet'), bin.density, 0)) AS pct_mobile_4g
FROM 
  `chrome-ux-report.chrome_ux_report.201710`,
  UNNEST(first_contentful_paint.histogram.bin) AS bin
WHERE
  origin = 'https://www.google.com'
```

**Comment by Themanwithoutaplan** [comment]
I can't look at the linked report but I'm slightly sceptical on the distribution.

 I'd rather be behing the curve on this, not least because I don't have 4G myself. The great advantage of the archive is that sites are tested on a like-for-like basis. In this case it doesn't really matter whether it's 3G or 4G but it's always good to aim for the lower. I'm finding more and more sites getting fatter and fatter (> 7MB is not unusual) and this is noticeable on mobile on a good wifi connection and downright unusable outside.

**Comment by pmeenan** [comment]
FWIW, my query also skews US since it doesn't look at all of the Google domains.  That said, every iteration I try is still > 85% 4G.

@Themanwithoutaplan we definitely want to skew towards the tail (I usually like to shoot for 70th-75th percentiles) but we need to make sure it also stays relevant.

The "effectively 4G" cutoff is also pretty slow.  Anything faster that 700Kbps with latency lower than 270ms is 4G.  There is a good amount of actual 3G that is faster than that (and actual 4G that can be slower than that).

**Comment by rviscomi** [comment]
> QQ. Since CrUX equalizes all origins, does that query basically average across all origins?

Yeah it's basically an average. (sum of all 4G densities over the sum of all densities for mobile)

> The great advantage of the archive is that sites are tested on a like-for-like basis. In this case it doesn't really matter whether it's 3G or 4G but it's always good to aim for the lower.

It's true that slower connections will dramatize performance issues, but I think we need to balance that with keeping up with changes in how real users access the web otherwise we lose relevance.

**Comment by paulcalvano** [comment]
The sampling does not seem to have a complete coverage of all connection types even for the most popular sites.   At a high level, for the 32,573 records, I found that crux mostly contains 3G and 4G connection types across desktop and Phone.

![image](https://user-images.githubusercontent.com/7459458/33961686-fb484f40-e01c-11e7-9532-e2a9039e2148.png)

The query for that was 
```
SELECT form_factor.name formFactor, effective_connection_type.name connType, COUNT(DISTINCT origin) freq
FROM `chrome-ux-report.chrome_ux_report.201710`
GROUP BY formFactor, connType
```

I looked for sites that contained all 10 form-factor / connection-types - I found only a handful of sites.   I joined Alexa rank (from the _pages table) and counted the number of histograms present for each form-factor and effective-connection-type using the following query:

```
SELECT rank, origin, form_factor.name formFactor, effective_connection_type.name connType, COUNT(DISTINCT origin) freq
FROM `chrome-ux-report.chrome_ux_report.201710` ux
JOIN  `httparchive.runs.2017_10_01_pages` pages
ON NET.HOST(pages.url) = NET.HOST(ux.origin)
WHERE rank > 0
GROUP BY rank, origin, formFactor, connType
ORDER By rank ASC
LIMIT 1000```

When I summarized it, I can see some gaps in the data for very large sites - especially with the tablet form-factor.
![image](https://user-images.githubusercontent.com/7459458/33961910-b532d8a8-e01d-11e7-8daa-6d2266d89990.png)

Should crux have histograms for all connection types and form-factors for the 10K sites included in the dataset, or is the selection of form-factors and connection-types to include a random sampling?

**Comment by paulcalvano** [comment]
Thanks @igrigorik.  That makes sense.

**Comment by igrigorik** [comment]
*bump*

**Comment by igrigorik** [comment]
Hmm, agreed, it would be nice to have consistent treatment. @paulirish @vinamratasingal any updates on LH roadmap for this?

**Comment by vinamratasingal** [comment]
Hey Ilya- @paulirish and @patrickhulce did a bunch of investigating and found some different conclusions re: 3G vs 4G from the ChrUX dataset... I think Patrick wrote up a doc, perhaps he can share and we can have a meeting to discuss? WDYT?

**Comment by igrigorik** [comment]
sg, curious to see the results

**Comment by rviscomi** [comment]
@pmeenan I think this is done. Can we close it?

**Comment by rviscomi** [comment]
I think we're still configuring connections to cable speeds. Leaving this issue open and transferring to the new repo.
